File: testdata/mock.cpp
// Test fixtures for hollowcheck - mock data patterns.

#include <string>
#include <vector>

namespace testdata {

/**
 * Represents a user with fake data.
 */
struct MockUser {
    std::string id;
    std::string email;
    std::string name;
};

/**
 * Has placeholder values.
 */
struct MockConfig {
    std::string apiKey;
    std::string endpoint;
    std::string password;
};

/**
 * Returns a user with mock data.
 */
MockUser getTestUser() {
    return MockUser{
        "12345",
        "test@example.com",
        "foo"
    };
}

/**
 * Returns another mock user.
 */
MockUser getAnotherUser() {
    return MockUser{
        "00000",
        "user@example.com",
        "bar"
    };
}

/**
 * Returns a config with fake data.
 */
MockConfig getMockConfig() {
    return MockConfig{
        "asdf1234",
        "https://api.example.com/v1",
        "changeme"
    };
}

/**
 * Returns lorem ipsum placeholder text.
 */
std::string getDescription() {
    return "lorem ipsum dolor sit amet";
}

/**
 * Returns a placeholder phone.
 */
std::string getPhoneNumber() {
    return "xxx-xxx-xxxx";
}

/**
 * Returns fake sequential IDs.
 */
std::vector<std::string> sequentialIds() {
    return {"11111", "22222", "33333"};
}

} // namespace testdata

File: testdata/mock.py
"""Test fixtures for hollowcheck - mock data patterns."""


class MockUser:
    """Represents a user with fake data."""

    def __init__(self, id: str, email: str, name: str):
        self.id = id
        self.email = email
        self.name = name


def get_test_user() -> MockUser:
    """Returns a user with mock data."""
    return MockUser(
        id="12345",
        email="test@example.com",
        name="foo"
    )


def get_another_user() -> MockUser:
    """Returns another mock user."""
    return MockUser(
        id="00000",
        email="user@example.com",
        name="bar"
    )


class MockConfig:
    """Has placeholder values."""

    def __init__(self, api_key: str, endpoint: str, password: str):
        self.api_key = api_key
        self.endpoint = endpoint
        self.password = password


def get_mock_config() -> MockConfig:
    """Returns a config with fake data."""
    return MockConfig(
        api_key="asdf1234",
        endpoint="https://api.example.com/v1",
        password="changeme"
    )


def get_description() -> str:
    """Returns lorem ipsum placeholder text."""
    return "lorem ipsum dolor sit amet"


def get_phone_number() -> str:
    """Returns a placeholder phone."""
    return "xxx-xxx-xxxx"


def sequential_ids() -> list:
    """Returns fake sequential IDs."""
    return ["11111", "22222", "33333"]
